core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/helper/SelectionHelper.java

```diff
@@ -50,6 +50,26 @@ public Collection<DDiagramElement> getElementsOfSameType(IStructuredSelection se
         .getDiagramContainer(semanticDecorators.get(0));
     return DiagramServices.getDiagramServices().getViewsRepresentingSameEType(diagram, semanticDecorators);
   }
+  
+  public Collection<DDiagramElement> getRelatedFCElements(IStructuredSelection selection) {
+    List<DSemanticDecorator> semanticDecorators = getDSemanticDecoratorsFromSelection(selection);
+    if (semanticDecorators.isEmpty() || semanticDecorators.size() != selection.size())
+      return new ArrayList<DDiagramElement>();
+
+    DSemanticDiagram diagram = (DSemanticDiagram) DiagramHelper.getService()
+        .getDiagramContainer(semanticDecorators.get(0));
+    return DiagramServices.getDiagramServices().getRelatedFCElements(diagram, semanticDecorators.get(0));
```

Comment:
can you make it work if you select several chains/paths at once

Reply:
Done